daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -573,8 +573,7 @@ private[lf] case class PartialTransaction(
           nodes = nodes.updated(nodeId, exerciseNode),
           actionNodeSeeds = actionNodeSeeds :+ (nodeId -> ec.actionNodeSeed),
         )
-      case _ =>
-        noteAbort(Tx.NonExerciseContext)
+      case _ => throw new RuntimeException("endExercises called in non-exercise context")
```

Comment:
@remyhaemmerle-da do you think we should throw something else here? `RuntimeException` is what the two other cases internal errors in `PartialTransaction` throw atm so this seems consistent. `crash` is not used at all here.

But happy to switch to something else or define a new exception if you prefer.

Reply:
I am fine with `RuntimeException`.